luigi/worker.py

```diff
@@ -789,7 +802,20 @@ def _get_work(self):
                 self._scheduled_tasks.get(batch_id) for batch_id in r['batch_task_ids']])
             self._batch_running_tasks[task_id] = batch_tasks
 
-        return task_id, running_tasks, n_pending_tasks, n_unique_pending, worker_state
+        r.setdefault('worker_state', WORKER_STATE_ACTIVE)
+        r.setdefault('n_pending_last_scheduled', 0)
+
+        return GetWorkResponse(
+            task_id=task_id,
+            running_tasks=running_tasks,
+            n_pending_tasks=r['n_pending_tasks'],
+            n_unique_pending=r['n_unique_pending'],
+
+            # TODO: For a tiny amount of time (a month?) we'll keep forwards compatibility
+            #  That is you can user a newer client than server (Sep 2016)
+            n_pending_last_scheduled=r.get('n_pending_last_scheduled', 0),
+            worker_state=r.get('worker_state', 0),
```

Comment:
Shouldn't `0` be `WORKER_STATE_ACTIVE`? And why is there both `get()` and `setdefault`?